File: src/p0014/golang/solution.go
func longestCommonPrefix(v []string) string {
	if len(v) == 0 {
		return ""
	}
	result := []byte(nil)
	for i := 0; ; i++ {
		for j := 0; j < len(v); j++ {
			if i >= len(v[j]) || v[j][i] != v[0][i] {
				return string(result)
			}
		}
		result = append(result, v[0][i])
	}
}

File: src/p0014/rust/solution.rs
impl Solution {
    pub fn longest_common_prefix(v: Vec<String>) -> String {
        let mut result = String::new();
        if v.is_empty() { return result; }
        let mut i: usize = 0;
        loop {
            for j in 0..v.len() {
                if i >= v[j].len() || (v[j].as_bytes()[i] != v[0].as_bytes()[i]) {
                    return result;
                }
            }
            result.push(v[0].as_bytes()[i] as char);
            i += 1;
        }
    }
}
